### ISSUE TITLE
Is project dead?

### ISSUE THREAD

**Issue by XVilka** [opensourcerouting/c-capnproto#55]
PRs aren't being reacted upon, last commit two years ago. Would be nice to make it work on modern systems and make a new release.

**Comment by eqvinox** [comment]
Yes it's dead. If you're interested in spending time on maintenance (merging PRs, making releases, etc.) I'm happy to give anyone access to the project here. Or you can just fork it; I'm happy to add a pointer to a new fork (after a little time to see/confirm that the new fork is in fact active and not equally dead.)

**Comment by eqvinox** [comment]
Background: we [my employer] originally had a project that used c-capnproto, and we invested time into maintaining it. However, that project died off, and I don't have a personal use case for c-capnproto either (even though I strongly prefer Cap'n'Proto over e.g. Protobuf), so… well… sorry, but it's just not on my plate anymore :disappointed:

**Comment by XVilka** [comment]
@eqvinox these are painful news. I would have loved to help, but I am already stretched too thin between multiple FOSS projects. Sadly, I don't have time to maintain this. I recommend to create and pin the issue that the project is searching for maintainers/developers.

**Comment by eqvinox** [comment]
> @eqvinox these are painful news. I would have loved to help, but I am already stretched too thin between multiple FOSS projects.

Yep… all too common of a situation :disappointed:

> Sadly, I don't have time to maintain this. I recommend to create and pin the issue that the project is searching for maintainers/developers.

Pinned. Also updated the README and just clicked merge on all PRs (without really looking at them TBH.)

Invited you for repo access in case you want to hit merge if you have follow-up meson/CI fixes.

**Comment by eqvinox** [comment]
K, threw an hour at it, fixed some random shit, time to go disappear for another year… o7

**Comment by XVilka** [comment]
@eqvinox thank you! I will try to fix few more things, improve CI and make a new release while I am on it.

**Comment by amery** [comment]
> @eqvinox thank you! I will try to fix few more things, improve CI and make a new release while I am on it.

could we have 0.4 tagged?

**Comment by jonahbeckford** [comment]
FYI: I've forked c-capnproto, made several fixes, added support for MSVC and improved the test/build matrix. I'm not planning on submitting PRs to this project since the maintainer did say this was unmaintained.

https://gitlab.com/dkml/ext/c-capnproto

This project is pretty essential. Thanks @eqvinox for taking it to this point!

**Comment by XVilka** [comment]
@jonahbeckford if you send small PRs I have time to review them, I can find a bit of time for that.

**Comment by XVilka** [comment]
Alternatively you could just ask capnproto update their site to point to your fork. This is probably easier for everyone.

**Comment by jonahbeckford** [comment]
@XVilka If I don't hear any contrary feedback on this issue, I'll ask capnproto to update their site in a few weeks.

In the meantime, I've already incorporated the changes into my fork, but if you or anyone has bandwidth to review them, just browse https://gitlab.com/dkml/ext/c-capnproto/-/commits/main?ref_type=heads and make an issue for anything that seems like it needs to change. Every commit is small except one: the "Add CMake and tests" commit is big but is 98% new code.

**Comment by jonahbeckford** [comment]
@XVilka Also can add you as a Developer to that repository. Developers can approve PRs; you don't have to be a maintainer. Just put a message on https://gitlab.com/dkml/ext/c-capnproto/-/merge_requests/1 if you want (which will let me get your GitLab username).

**Comment by jonahbeckford** [comment]
The request to change the active repository for c-capnproto is at https://groups.google.com/g/capnproto/c/0pfEYKVNI3A

**Comment by XVilka** [comment]
Once the switch is completed, it's probably better to archive this repository and point to the new home in the README.

**Comment by HaoZeke** [comment]
I've been maintaining a fork as well, I won't promise much personal effort but I'll keep it green and working..
